file/src/main/java/org/apache/zeppelin/file/HDFSCommand.java

```diff
@@ -0,0 +1,151 @@
+import java.io.InputStreamReader;
+import javax.ws.rs.core.UriBuilder;
+import org.slf4j.Logger;
+
+/**
+ * Definition and HTTP invocation methods for all WebHDFS commands
+ *
+ * @author rajbains
+ *
+ */
+public class HDFSCommand {
+
+  /**
+   * Type of HTTP request
+   */
+  public enum HttpType {
+    GET,
+    PUT
+  }
+
+  /**
+   * Definition of WebHDFS operator
+   */
+  public class Op {
+    public String op;
+    public HttpType cmd;
+    public int minArgs;
+
+    public Op(String op, HttpType cmd, int minArgs) {
+      this.op = op;
+      this.cmd = cmd;
+      this.minArgs = minArgs;
+    }
+  }
+
+  /**
+   * Definition of argument to an operator
+   */
+  public class Arg {
+    public String key;
+    public String value;
+
+    public Arg(String key, String value) {
+      this.key = key;
+      this.value = value;
+    }
+  }
+
+  // How to connect to WebHDFS
+  String url = null;
+  String user = null;
+  Logger logger;
+
+  // Define all the commands available
+  public Op getFileStatus = new Op("GETFILESTATUS", HttpType.GET, 0);
+  public Op listStatus = new Op("LISTSTATUS", HttpType.GET, 0);
+
+  public HDFSCommand(String url, String user, Logger logger) {
+    super();
+    this.url = url;
+    this.user = user;
+    this.logger = logger;
+  }
+
+  public String checkArgs(Op op, String path, Arg[] args) throws Exception {
+    if (op == null ||
+        path == null ||
+        (op.minArgs > 0 &&
+            (args == null ||
+                args.length != op.minArgs)))
+    {
+      String a = "";
+      a = (op != null) ? a + op.op + "\n" : a;
+      a = (path != null) ? a + path + "\n" : a;
+      a = (args != null) ? a + args + "\n" : a;
+      return a;
+    }
+    return null;
+  }
+
+
+  // The operator that runs all commands
+  public String runCommand(Op op, String path, Arg[] args) throws Exception {
+
+    // Check arguments
+    String error = checkArgs(op, path, args);
+    if (error != null) {
+      logger.error("Bad arguments to command: " + error);
+      return "ERROR: BAD ARGS";
+    }
+
+    // Build URI
+    UriBuilder builder = UriBuilder
+        .fromPath(url)
+        .path(path)
+        .queryParam("op", op.op);
+
+    if (args != null) {
+      for (Arg a : args) {
+        builder = builder.queryParam(a.key, a.value);
+      }
+    }
+    java.net.URI uri = builder.build();
+
+    // Connect and get response string
+    URL hdfsUrl = uri.toURL();
+    HttpURLConnection con = (HttpURLConnection) hdfsUrl.openConnection();
+
+    if (op.cmd == HttpType.GET) {
+      con.setRequestMethod("GET");
+      int responseCode = con.getResponseCode();
+      logger.info("Sending 'GET' request to URL : " + hdfsUrl);
+      logger.info("Response Code : " + responseCode);
+
+      BufferedReader in = new BufferedReader(
+          new InputStreamReader(con.getInputStream()));
+      String inputLine;
+      StringBuffer response = new StringBuffer();
+
+      while ((inputLine = in.readLine()) != null) {
```

Comment:
you might want to limit the length of the output if it gets very long.
one approach that has been used is to have a interpreter config for the length limit.